core/src/stages/pick.cpp

```diff
@@ -1,63 +1,143 @@
-  : SerialContainer(name) {
+PickPlaceBase::PickPlaceBase(const std::string& name, const std::string& provider_stage_plugin_name,  bool is_pick)
+  : SerialContainer(name),
+  grasp_provider_class_loader_("moveit_task_constructor_core", "moveit::task_constructor::stages::GraspProviderBase"),
+  place_provider_class_loader_("moveit_task_constructor_core", "moveit::task_constructor::stages::PlaceProviderBase") {
 	PropertyMap& p = properties();
-	p.declare<std::string>("object", "name of object to grasp");
+	p.declare<std::string>("object", "name of object to grasp/place");
 	p.declare<std::string>("eef", "end effector name");
-	p.declare<std::string>("eef_frame", "name of end effector frame");
+	p.declare<geometry_msgs::PoseStamped>("ik_frame", "frame to be moved towards goal pose");
+	p.declare<std::string>("eef_group_open_pose", "Name of open pose of eef_group");
+	p.declare<std::string>("eef_group_close_pose", "Name of close pose of eef_group");
+	p.declare<std::vector<std::string>>("support_surfaces", {}, "Name of support surfaces");
 
 	// internal properties (cannot be marked as such yet)
 	p.declare<std::string>("eef_group", "JMG of eef");
 	p.declare<std::string>("eef_parent_group", "JMG of eef's parent");
 
 	cartesian_solver_ = std::make_shared<solvers::CartesianPath>();
-	int insertion_position = forward ? -1 : 0;  // insert children at end / front, i.e. normal or reverse order
+	sampling_planner_ = std::make_shared<solvers::PipelinePlanner>();
 
-	auto init_ik_frame = [](const PropertyMap& other) -> boost::any {
-		geometry_msgs::PoseStamped pose;
-		const boost::any& frame = other.get("eef_frame");
-		if (frame.empty())
-			return boost::any();
+	is_pick_ = is_pick;
 
-		pose.header.frame_id = boost::any_cast<std::string>(frame);
-		pose.pose.orientation.w = 1.0;
-		return pose;
-	};
+	provider_stage_plugin_name_ = provider_stage_plugin_name;
 
 	{
-		auto approach = std::make_unique<MoveRelative>(forward ? "approach object" : "retract", cartesian_solver_);
-		PropertyMap& p = approach->properties();
+		auto move_there = std::make_unique<MoveRelative>(is_pick_ ? "approach object" : "lower object", cartesian_solver_);
+		PropertyMap& p = move_there->properties();
 		p.property("group").configureInitFrom(Stage::PARENT, "eef_parent_group");
-		p.property("ik_frame").configureInitFrom(Stage::PARENT, init_ik_frame);
-		p.set("marker_ns", std::string(forward ? "approach" : "retract"));
-		approach_stage_ = approach.get();
-		insert(std::move(approach), insertion_position);
+		p.property("ik_frame").configureInitFrom(Stage::PARENT, "ik_frame");
+		p.set("marker_ns", std::string(is_pick_ ? "approach" : "place"));
+		move_there_stage_ = move_there.get();
+		insert(std::move(move_there));
+	}
+
+	if (provider_stage_plugin_name_ == "")
+	{
+		if (is_pick_)
+			provider_stage_plugin_name_ = "moveit_task_constructor/GraspProviderDefault";
+		else
+			provider_stage_plugin_name_ = "moveit_task_constructor/PlaceProviderDefault";
```

Comment:
I'd call it `PlacePoseProviderDefault` if that hasn't already been decided somewhere else.